### PR TITLE
chown the mounted volume after generation

### PR DESCRIPTION
**Pull request by bombela**
For `make generate` with Docker we mount the local workdir withing the
container. Any files written from withing the container will be owned by
whatever user is loggued in (alwaysr root in our case). The container is
already root on the workdir. We can the chown recursively the workdir to the
owner of the workdir.

The chown only runs withing the docker container. For this to work, a new env
var `WITHIN_DOCKER` is passed to the container.

Note: for what I know. On Mac, you run a linux virtual machine, which in turns
runs containers. You are loggued as root on the linux VM, so you have no
resstrictions. And the virtual machine shares the filesystem between Mac and
the linux VM by rewriting on the fly permissions from the owner of the VM to
root.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -15,6 +15,7 @@ TAGS
 encoding/thrift/thriftrw-plugin-yarpc/thriftrw-plugin-yarpc
 encoding/x/protobuf/protoc-gen-yarpc-go/protoc-gen-yarpc-go
 internal/crossdock/crossdock
+internal/examples/env/
```
Comment (peter-edge) [inline .dockerignore]: Remove this in .gitignore - you probably have a clone from a long time ago, this is no longer generated.

File: .gitignore [hunk 2]
```diff
@@ -15,6 +15,7 @@
 /encoding/thrift/thriftrw-plugin-yarpc/thriftrw-plugin-yarpc
 /encoding/x/protobuf/protoc-gen-yarpc-go/protoc-gen-yarpc-go
 /internal/crossdock/crossdock
+/internal/examples/env/
```
Comment (peter-edge) [inline .gitignore]: Remove

File: build/docker.mk [hunk 3]
```diff
@@ -5,7 +5,8 @@ DOCKER_IMAGE := uber/yarpc-go-$(DOCKER_GO_VERSION)
 ifdef DOCKER_HOST
 DOCKER_BUILD_FLAGS ?= --compress
 endif
-DOCKER_RUN_FLAGS ?= -e V -e RUN -e EXAMPLES_JOBS
+DOCKER_RUN_FLAGS ?= -e V -e RUN -e EXAMPLES_JOBS -e WITHIN_DOCKER=1
+DOCKER_VOLUMES=-v $(shell pwd):/go/src/go.uber.org/yarpc
```
Comment (peter-edge) [inline build/docker.mk]: Rename to `DOCKER_VOLUME_FLAGS`? `DOCKER_VOLUMES` would be without the `-v`.

Comment (bombela) [inline build/docker.mk]: fair enough

**Pull request merged**